csharp/Svix/SvixClient.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Svix.Abstractions;
+using Svix.Api;
+using Svix.Client;
+using Svix.Models;
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Svix
+{
+    public sealed class SvixClient : ISvixClient
+    {
+        private readonly ILogger _logger;
+        
+        private readonly SvixClientOptions _options;
+
+        public SvixClient(SvixClientOptions options, ILogger<SvixClient> logger = null)
+        {
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _logger = logger;
+        }
+
+        public SvixClient(IOptions<SvixClientOptions> options, ILogger<SvixClient> logger)
+            : this(options?.Value, logger)
+        {
+            // empty
+        }
+        
+        #region " Health "
+        
+        public bool IsHealthy(string idempotencyKey = default)
+        {
+            try
+            {
+                var lConfig = new Configuration
+                {
+                    BasePath = _options.ServiceUrl,
+                    AccessToken = _options.AccessToken
+                };
+                
+                using var lHealthApi = new HealthApi(lConfig);
+                var lResponse = lHealthApi.HealthApiV1HealthGetWithHttpInfo(idempotencyKey);
+
+                return lResponse.StatusCode == HttpStatusCode.NoContent;
+            }
+            catch (ApiException e)
+            {
+                _logger?.LogError(e, $"{nameof(IsHealthy)} failed");
+
+                if (_options.Throw)
+                    throw;
+                
+                return false;
+            }
+        }
```

Comment:
The implementation itself here looks great though to keep consistent with our other libraries we should define a class called `Health` and have it hold all the functions for the health api (and do the same for all other api tags `Message`, `Application`, etc.

That way the caller would call:
```
svixClient.Health.IsHealthy()
```
```
svixClient.Message.Create(...)
```
etc.

Doing this would also allow us to construct the internal generated api classes once when we construct the SvixClient as well to avoid additional overhead when making the calls